packages/OntologyService/__tests__/OntologyService.fn.test.ts

```diff
@@ -0,0 +1,75 @@
+import { OntologyService, QueryResponse, TypedNodeQuerySolution } from "../index"
+
+
+// create a successful sparql result object - as expected to return
+// can check this by generating this data and sending queries to a database offline 
+// but we know the good response for this query would look like this
+
+const uri1:string = "http://telicent.io/fake_data#uri1"
+const MOCK_URL = "http://localhost:3030/"
+const os = new OntologyService(MOCK_URL)
+
+const RESPONSE_getAllClasses_uri1 = {
+    head: { vars: ["uri", "_type"] },
+   results: {   
+    bindings:[{
+        "uri": {"type":"uri", "value":uri1},
+        "_type": {"type": "uri", "value": "http://www.w3.org/1999/02/22-rdf-syntax-ns#Class"}
+    }]
+   }
+} as unknown as Promise<QueryResponse<TypedNodeQuerySolution>>
+
+
+
+describe("testing functions within the ontology service", ()=>{
+    beforeEach(() => {
+        jest.resetAllMocks()
+    })
+  /*  it("new RDFSClass - should produce correct SPARQL update", async () => {
+        const runUpdateMock = jest.spyOn(os, "runUpdate").mockImplementation(()=> Promise.resolve("OK"))
+        const CORRECT_QUERY = "SELECT ?uri ?_type WHERE {BIND (rdfs:Class as ?_type . ?uri a ?_type ) . }"
+        // run function
+        const res = await os.getAllClasses(false)
+        // check correct query is passed based on function call to runQuery
+        expect(runUpdateMock).toHaveBeenCalledWith(CORRECT_QUERY)
+    });*/
+    it("getAllClasses - check correct query is created and sent to runQuery", async () => {
+        // runQuery should be tested as part of the RDFService tests - return SUCCESSFULL_RESPONSE
+        const runQueryMock = jest.spyOn(os, "runQuery").mockImplementation(()=> Promise.resolve(RESPONSE_getAllClasses_uri1))
+        // this is the correct query - no filter and not including owl
+        const CORRECT_QUERY = "SELECT ?uri (group_concat(DISTINCT ?type) as ?_type) WHERE {BIND (rdfs:Class as ?type . ?uri a ?type ) . } GROUP BY ?uri"
```

Comment:
Probably immaterial, but fyi [this tool](https://query.wikidata.org/#SELECT%20%3Furi%20%28group_concat%28DISTINCT%20%3Ftype%29%20as%20%3F_type%29%20WHERE%20%7BBIND%20%28rdfs%3AClass%20as%20%3Ftype%20.%20%3Furi%20a%20%3Ftype%20%29%20.%20%7D%20GROUP%20BY%20%3Furi) has a problem with `?type . ?uri`